adapters/tlsio_openssl.c

```diff
@@ -897,6 +897,226 @@ static int load_cert_crl_http(
     return rv;
 }
 
+static int atoin(const char *str, int start, int len)
+{
+    int result = 0;
+
+    for (; len > 0; len--, start++)
+    {
+        if (str[start] < '0' || str[start] > '9')
+            return -1;
+
+        result = (result * 10) + (str[start] - '0');
+    }
+
+    return result;
+}
+
+static time_t crl_invalid_after(X509_CRL *crl)
+{
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+    const ASN1_TIME *at = X509_CRL_get0_nextUpdate(crl);
+#else
+    ASN1_TIME *at = crl->crl->nextUpdate;
+#endif
+
+    ASN1_GENERALIZEDTIME *gt = ASN1_TIME_to_generalizedtime(at, NULL);
```

Comment:
what if "at" is null? does ANS1 handles this gracefully?

Reply:
I would hope so